### PR TITLE
Fix: change error code in function isAccessDenied to AccessDenied

### PR DESCRIPTION
**Pull request by blackpiglet**
Signed-off-by: Xun Jiang <blackpiglet@gmail.com>

What does this PR change? What problem does it solve?
-----------------------------------------------------

It's used to fix issue #4085

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: internal/backend/s3/s3.go [hunk 1]
```diff
@@ -164,7 +164,7 @@ func isAccessDenied(err error) bool {
 	debug.Log("isAccessDenied(%T, %#v)", err, err)
 
 	var e minio.ErrorResponse
-	return errors.As(err, &e) && e.Code == "Access Denied"
+	return errors.As(err, &e) && e.Code == "AccessDenied"
```
Comment (greatroar) [inline internal/backend/s3/s3.go:167]: LGTM. Sorry for screwing this up.

File: changelog/unreleased/issue-4085 [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix error code in function isAccessDenied.
```
[the remark this answers is not in the document]
Comment (blackpiglet) [reply changelog/unreleased/issue-4085:1]: Thanks for the guidance. That's much better.
Accept the suggestion, but need to make a little change in description. It should be `check bucket existence`, not `create a bucket`.

File: internal/backend/s3/s3.go (hunk 1 above)
Comment (blackpiglet) [reply internal/backend/s3/s3.go:167]: Not a problem.
I'm glad to give some contribution.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Comment by MichaelEischer** [comment]
I've rebased the PR and slightly shortened the title of the changelog entry to fix the test failure.

**Pull request merged**